scripts/test-e2e-local.js

```diff
@@ -147,53 +166,75 @@ if (argv.target === 'RNTester') {
+  // Generate native files for Android
   generateAndroidArtifacts(releaseVersion, tmpPublishingFolder);
 
+  // Setting up generating native iOS (will be done later)
+  const repoRoot = pwd();
+  const jsiFolder = `${repoRoot}/ReactCommon/jsi`;
+  const hermesUtilsFolder = `${repoRoot}/sdks/hermes-engine/utils`;
+  const hermesCoreSourceFolder = `${repoRoot}/sdks/hermes`;
+
+  if (!fs.existsSync(hermesCoreSourceFolder)) {
+    console.info('The Hermes source folder is missing. Downloading...');
+    downloadHermesSourceTarball();
+    expandHermesSourceTarball();
+  }
+
+  // for this scenario, we only need to create the debug build
+  // (env variable PRODUCTION defines that podspec side)
+  const buildType = 'Debug';
+
+  // the android ones get set into /private/tmp/maven-local
+  const localMavenPath = '/private/tmp/maven-local';
+
+  // Generate native files for iOS
+  const tarballOutputPath = generateiOSArtifacts(
+    jsiFolder,
+    hermesUtilsFolder,
+    hermesCoreSourceFolder,
+    buildType,
+    releaseVersion,
+    localMavenPath,
+  );
+
   // create locally the node module
   exec('npm pack');
 
-  const localNodeTGZPath = `${pwd()}/react-native-${releaseVersion}.tgz`;
+  const localNodeTGZPath = `${repoRoot}/react-native-${releaseVersion}.tgz`;
   exec(`node scripts/set-rn-template-version.js "file:${localNodeTGZPath}"`);
 
-  const repoRoot = pwd();
-
   pushd('/tmp/');
-  // need to avoid the pod install step because it will fail! (see above)
+  // need to avoid the pod install step - we'll do it later
   exec(
     `node ${repoRoot}/cli.js init RNTestProject --template ${repoRoot} --skip-install`,
   );
 
   cd('RNTestProject');
   exec('yarn install');
 
+  // need to do this here so that Android will be properly setup either way
+  exec(
+    'echo "REACT_NATIVE_MAVEN_LOCAL_REPO=/private/tmp/maven-local" >> android/gradle.properties',
+  );
+
   if (argv.platform === 'iOS') {
     // if we want iOS, we need to do pod install - but with a trick
     cd('ios');
     exec('bundle install');
 
-    // TODO: we should be able to also use HERMES_ENGINE_TARBALL_PATH
-    // if we can make RNTester step generate it already so that it gets reused
-
-    // need to discern if it's main branch or release branch
-    if (baseVersion === '1000.0.0') {
-      // main branch
-      exec(`USE_HERMES=${argv.hermes ? 1 : 0} bundle exec pod install --ansi`);
-    } else {
-      // TODO: to test this, I need to apply changes on top of a release branch
-      // a release branch
-      // copy over the .hermesversion file from react-native core into the RNTestProject
-      exec(`cp -f ${repoRoot}/sdks/.hermesversion .`);
-      exec(
-        `CI=true USE_HERMES=${
-          argv.hermes ? 1 : 0
-        } bundle exec pod install --ansi`,
-      );
-    }
+    // set HERMES_ENGINE_TARBALL_PATH to point to the local artifacts I just created
+    exec(
+      `HERMES_ENGINE_TARBALL_PATH=${tarballOutputPath} USE_HERMES=${
```

Comment:
this part I don't particularly like because it means that if the tester needs to test something else with RNTestProject - ex. test that everything still works on JSC, they will have to do a new pod install without hermes, but if after that they want to test Hermes again they will have to manually pass this `HERMES_ENGINE_TARBALL_PATH` variable during the pod install :(

Sadly because of limitations of shelljs we can't set ENV variables through it, and since there's no `gradle.properties` equivalent we're stuck with this. Hopefully it won't create many attrictions